lib/irc.js

```diff
@@ -790,6 +799,29 @@ Client.prototype.connect = function(retryCount, callback) {
         self.emit('netError', exception);
     });
 };
+
+Client.prototype.enableParseMiddleware = function(middleware) {
+    if (middleware && typeof middleware === 'function') {
+        this._parseMiddlewares.push(middleware);
+    }
+};
+
+Client.prototype.disableParseMiddleware = function(middleware) {
+    if (middlware && typeof middleware === 'function') {
+        this._parseMiddlewares = this.parseMiddlewares.filter(function(func) {
+            return func !== middleware;
```

Comment:
Preserving the order that the middlewares were added is important, so an array made sense to me.